daemons/ipa-kdb/ipa_kdb_principals.c

```diff
@@ -774,6 +774,15 @@ static krb5_error_code ipadb_parse_ldap_entry(krb5_context kcontext,
                                       &res_key_data, &result, &mkvno);
     switch (ret) {
     case 0:
+        /* Only set a principal's key if password auth should be used to make
+         * sure that the KDC relies do not advertise password auth if not
+         * available. */
```

Comment:
Thank you for expanding this comment from its previous version.  Could you perhaps mention more of what your commit message contained about the responder interface and OTP?  It's still not clear from the comment what will break if this conditional were to be removed again (though of course it is from your commit message).